lib/order.rb

```diff
@@ -0,0 +1,58 @@
+class Order
+
+  attr_reader :dishes, :total, :calc, :menu
+
+  def initialize(dish_class = Dish)
+    @dish_class = dish_class
+    @dishes = []
+    @dish = nil
+    @total = 0
+    @menu = MENU
+  end 
+
+  def add_dish(item, quantity = 1) 
+    
+    quantity.times do
+      @dish = dish_creator(item)
+      accumulator(@dish)
+      @dishes << @dish
+      @dish = nil
+    end
```

Comment:
I don't think you need `@dish` to be an instance variable here. You can return it from dish_creator and use it as a local variable instead.

```rb
      dish = dish_creator(item)
      accumulator(dish)
      @dishes << dish
```

Wherever possible, you should avoid storing data in instance variables.